### DISCUSSION TITLE
Settings for MS2109 based grabber

### DISCUSSION THREAD

**Discussion by hkalle** [awawa-dev/HyperHDR#294] [Q&A]
Hello Awawa

Sorry for posting here and not direclty in your blog (https://www.hyperhdr.eu/2022/04/usb-grabbers-hdr-to-sdr-quality-test.html?lr=1). I was not able to activate the comment function.

Thanks a lot for this interesting analysis.

I am using a FeinTech VSP01222 splitter and a MS2109 based grabber. Also followed your recommendation for the grabber settings and implemented the optimized LUT table. Could please point me into the right direction regarding two topics:

1. I understood, that the settings for the MS2109 chip is for HDR grabbing. Do I still need to implement another profil & LUT table for SDR grabbing, when using YUYV?
2.  Independent from the HDR topic the picture deliverd by the grabber is in general to greenish and I tried to compensate via the Hue, which was not very successful. Any hint, what I still could try?

Thanks upfront!

**Comment by awawa-dev** [comment]
HI
Which MS2109 based grabber you are using and what OS? Some version of MS2109 grabbers are known to provide worse quality picture due to some modification.

**Comment by hkalle** [comment]
Hi 

I tried two:

In use
https://www.amazon.de/gp/product/B08FDM6N6H/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
Same behaviour with this one
https://www.amazon.de/gp/product/B08DFRS36T/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

**Comment by awawa-dev** [comment]
So far I've got reports about some problems with fake USB3.0 versions, not sure about winfox implementation (it's USB2.0). Maybe it's due to video signal properties? If you set proper saturation/contrast/brightness for MS2109 and that didn't help then maybe try to calibrate your grabber (1080p 10-bit as a reference, 30 video refresh rate or more if available)? Immediately after the calibration is finished, please take a look to your logs. There should be more information about how the calibration went and the final score.

**Comment by marcosvrs** [comment]
If I can be of any help, I ordered this one: https://www.berrybase.de/usb-hdmi-video-grabber-1080p
From what I could see until now, the following configs worked for me:
Frames per second: 60 (Otherwise it gets too laggy)
Grabbers video format: mjpeg (yuyv didn't work well for me)
Brightness: 1
Contrast: 128
Saturation: 128
Hue: 1

I use the LUT provided on the post you mentioned first. I also tried to use the v18beta to auto-generate the LUT, but it ends up being the same result.

**Reply by marcosvrs** [reply]
> Independent from the HDR topic the picture deliverd by the grabber is in general to greenish and I tried to compensate via the Hue, which was not very successful. Any hint, what I still could try?

This is also occurring on my setup BTW, I don't know if there is a known solution for it, though.
![shot223017](https://user-images.githubusercontent.com/10147574/178352730-67359d09-b976-40dd-a693-53334102a0d8.png)

**Reply by awawa-dev** [reply]
>  I also tried to use the v18beta to auto-generate the LUT, but it ends up being the same result.

Again, there are no calibration logs so I don't know what's going on here, for example there are MS2109 implementations difference with each other. You can manually adjust the green gamma channel in the option before starting the calibration.

**Reply by marcosvrs** [reply]
Just to update here, I wasn't able to add more logs because my MS2109 died. Survived for less than 2 weeks RIP :(
I'm back with my ezcap269.

**Reply by awawa-dev** [reply]
If an electronic device is going to break down, it does at the beginning. But for grabbers usually it's caused by bad design and overheating. The build quality of the various ms2109 varies greatly and QC is low. Often, manufacturers cut costs and do not use internal heat sinks that should be there. This grabber seems very thin and I don't even know if it would fit there.  

Old barrel type grabber USB2.0 version:  
![obraz](https://user-images.githubusercontent.com/69086569/179396877-377c984d-0768-4fdf-9e76-8e4f613fbf0f.png)
